### PR TITLE
Fix issues with redirecting to catalog service provider

### PR DESCRIPTION
**Pull request by alanking**
Addresses #47

Confirmed redirect working by hand. Because no topology tests for plugins. The tests which do exist are all passing.

I added another redirect in the `pam_auth_agent_request` operation because I noticed that the catalog service provider was not being configured (specifically, the log level was not being set). I kept it in a separate commit in case we wanted to do that later or in a separate issue. Probably will still keep it in a separate commit even if we decide to keep it in this PR / issue.

### CODE REVIEW

File: plugin/src/main.cpp [hunk 1]
```diff
@@ -599,7 +593,10 @@ namespace irods
                           "and sensitive user information is being communicated over the network in an unencrypted "
                           "manner. Configure this server to require TLS/SSL to prevent security leaks.");
         }
-        return irods_auth::request(*host->conn, req);
+          // Note: We should not disconnect this server-to-server connection because the connection is not owned by
+          // this context. A set of server-to-server connections is maintained by the server agent and reused by
+          // various APIs and operations as needed.
+          return irods_auth::request(*host->conn, req);
```
Comment (trel) [inline plugin/src/main.cpp:645]: indentation?   ignore if correct.

**Comment by trel** [comment]
"shceme" in second commit message body

File: plugin/src/main.cpp [hunk 2]
```diff
@@ -528,35 +528,76 @@ namespace irods
...
+            // Note: We should not disconnect this server-to-server connection because the connection is not owned by
+            // this context. A set of server-to-server connections is maintained by the server agent and reused by
+            // various APIs and operations as needed.
+            return irods_auth::request(*host->conn, req);
         }
 
-        comm.auth_scheme = strdup(pam_interactive_scheme);
-        return resp;
+        // This operation does not have anything to do add to the JSON structure, so just copy the request structure.
+        return {req};
```
Comment (korydraughn) [inline plugin/src/main.cpp:591]: I think the curly braces can be dropped because `req` is const.

File: plugin/src/main.cpp [hunk 3]
```diff
@@ -528,35 +528,76 @@ namespace irods
...
+            }
+            // Note: We should not disconnect this server-to-server connection because the connection is not owned by
+            // this context. A set of server-to-server connections is maintained by the server agent and reused by
+            // various APIs and operations as needed.
+            return irods_auth::request(*host->conn, req);
         }
 
-        comm.auth_scheme = strdup(pam_interactive_scheme);
-        return resp;
+        // This operation does not have anything to do add to the JSON structure, so just copy the request structure.
```
Comment (trel) [inline plugin/src/main.cpp:590]: "to do add"
to
"to add"

**Review by korydraughn — COMMENTED** [review]
Looks good to me.

File: plugin/src/main.cpp (hunk 1 above)
Comment (alanking) [reply plugin/src/main.cpp:645]: The indentation in this function is really messed up, so I'm just correcting indentation a little bit at a time as lines (or adjacent ones) are touched. So, it's not consistent with the rest of the function, but is more correct.

File: plugin/src/main.cpp (hunk 3 above)
Comment (alanking) [reply plugin/src/main.cpp]: Done

File: plugin/src/main.cpp (hunk 2 above)
Comment (alanking) [reply plugin/src/main.cpp]: Done

**Comment by korydraughn** [comment]
Please verify the tests still pass _(since the curly braces were removed. I'm feeling paranoid about that change right now lol)_.

**Comment by alanking** [comment]
Tests are still passing, yep

**Comment by korydraughn** [comment]
Good. Squash it when ready.

**Comment by alanking** [comment]
Squashed and corrected "shceme" typo in second commit message.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by alanking** [comment]
#'d, mergin

**Pull request merged**